### ISSUE TITLE
Crash vita2d_load_PNG_file in VitaShell and other app

### ISSUE THREAD

**Issue by rozenmad** [xerpi/vita2dlib#62]
I found a bug with loading some png files, it also happens in vitashell and my own written applications. I tried to find the reason and came to the conclusion that perhaps the reason is allocation memory for texture. It crashes on **png_read_image**.
<details>

![Picture](https://user-images.githubusercontent.com/7805174/44006154-108c384c-9e88-11e8-91b8-2bf0e8ec444a.png)

![Picture](https://user-images.githubusercontent.com/7805174/44006153-106b168a-9e88-11e8-875f-9ada283b7473.png)
</details>

But if you allocate memory more than the original texture size in **_vita2d_create_empty_texture_format_advanced**, then there is no error.

Bug: http://rextester.com/edit/IVQYW44809

**Comment by BenMitnicK** [comment]
Maybe I have the same issue that you

**Comment by xerpi** [comment]
How many images are you loading and how large are they?

EDIT: I'll do some tests

**Comment by BenMitnicK** [comment]
For me 64 for my theme manager and over 100 for reading books

Max 900K and for books apps I don't know if people make their image books more than 1mo or not

**Comment by xerpi** [comment]
Can you try this patch? https://hastebin.com/raw/wepevewumu

To apply it run: `patch -p0 < file`

**Comment by BenMitnicK** [comment]
Yes but this weekend because I'm work outside my region

**Comment by BenMitnicK** [comment]
Your link is dead resend please :)

**Comment by xerpi** [comment]
https://pastebin.com/raw/9EL6yPZt

**Comment by BenMitnicK** [comment]
THX

it's a first for me this kind of operation, you can give me the procedure to follow. where paste the code and the command I enter it directly on the terminal ??

**Comment by xerpi** [comment]
Download it as patch.diff and then run `patch -p0 < patch.diff`

**Comment by BenMitnicK** [comment]
ok I use MSYS2 and commande patch not found I go on google for installing it if we can

**Comment by BenMitnicK** [comment]
I installed patch and apply patch I have this, say me if it's ok or not

$ patch -p0 < patch.diff
(Stripping trailing CRs from patch; use --binary to disable.)
patching file libvita2d/source/vita2d_texture.c
patch unexpectedly ends in middle of line
Hunk #4 succeeded at 232 with fuzz 1.

**Comment by BenMitnicK** [comment]
I compiled vita2d and try it and the same problem

**Comment by xerpi** [comment]
Hmm that's weird, I'll send a compiled version just in case:
[libvita2d.zip](https://github.com/xerpi/libvita2d/files/2704851/libvita2d.zip)
Btw make sure you first do `make clean` and then `make` on your application.

**Comment by BenMitnicK** [comment]
yes of course I clean before, I even deleted the contents of the directory to be on

I tested your file the same problem

Maybe it's my code array ... you watched it if the array it's not declared correctly ?

**Comment by xerpi** [comment]
It might be that you are loading too many images and thus filling all the memory. Can you paste your code?

**Comment by BenMitnicK** [comment]
look my issue [here](https://github.com/xerpi/libvita2d/issues/66)

**Comment by devnoname120** [comment]
Relevant: https://github.com/xerpi/libvita2d/issues/66#issuecomment-507862542
